src/cogent3/app/composable.py

```diff
@@ -904,12 +904,22 @@ def _repr(self):
 
 def _new(klass, *args, **kwargs):
     obj = object.__new__(klass)
-
-    init_sig = inspect.signature(klass.__init__)
-    bargs = init_sig.bind_partial(klass, *args, **kwargs)
+    if not args and not kwargs:
+        obj._init_vals = {}
+        return obj
+
+    if hasattr(klass, "_func_sig"):
+        # we have a decorated function, the first parameter in the signature
+        # is not given to constructor, so we create a new signature excluding that one
+        params = klass._func_sig.parameters
+        init_sig = inspect.Signature(parameters=list(params.values())[1:])
+        bargs = init_sig.bind_partial(*args, **kwargs)
```

Comment:
If we have a user function, does this work? because user function does not have `self`?

Reply:
the bound function cannot access self, but the methods on the containing class can

Reply:
So, does the starting index need to be zero, for decorated user function?